### PR TITLE
Hr events

### PR DESCRIPTION
**Pull request by HakonRydland**
## What type of PR is this.
1. [ ] Bug
2. [ ] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    Final part of 3rd party support:
    This part adds the event system to antistasi whit one example usage of event trigger, with this we could allow 3rd party mods to extend (in this case) AIVehInit to add functionality to the gamemode

Slightly out of scope changes:
   - Added `Warning` log macros
   - Added macros for detecting if addon is loaded (for compatibility instead of dependancy in the future)
   - Fixed some broken (but unused) macros
### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [x] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [x] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps:  
- add an event listener like so 
```sqf
["AIVehInit", "test", {systemChat str _this}] call A3A_Events_fnc_addEventListener;
```
- now test it by buying a vehicle
- then remove one like so and retest
```sqf
["AIVehInit", "test"] call A3A_Events_fnc_removeEventListener;
```

- Then add two more and remove all for AIVehInit like so (test before and after removing)
```sqf
["AIVehInit", "test1", {systemChat "1"}] call A3A_Events_fnc_addEventListener;
["AIVehInit", "test2", {systemChat "2"}] call A3A_Events_fnc_addEventListener;

["AIVehInit"] call A3A_Events_fnc_removeAllEventListeners;
```

- Finally trigger a false event like so
```sqf
["AIVehInit", [player, side group player]] call A3A_Events_fnc_triggerEvent;
```
********************************************************
Notes:
This only adds the framework and an example event, there would need to be added a lot more events before it would be of substantial use to 3rd party developers.

### CODE REVIEW

**Comment by Bob-Murphy** [comment]
Is this ready for review?

**Comment by HakonRydland** [comment]
Yes, core mechanics can be reviewed, still need to decide and create all the hooks, but that can be done at a later point

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]
typos and such

**Review by killerswin2 — APPROVED** [review]
LGTM, pray for events

File: A3A/addons/Events/Events.hpp [hunk 1]
```diff
@@ -1,20 +1,159 @@
...
+    class AIVehInit {
+        /*
+            This Event registry shows that the event expects two arguments, the first should be an object, the second should be a side
+            both arguments are required in this case
+        */
+        isLocal = 1; //flag to show that this function is only triggered on local machine (for end user)
+        class params {
+            // naming scheme of elements: _{Index}
+            class _0 {
+                description = "Vehicle being initilized";
```
Comment (Bob-Murphy) [inline A3A/addons/Events/Events.hpp:11]: initialized

File: A3A/addons/Events/Events.hpp [hunk 2]
```diff
@@ -1,20 +1,159 @@
...
+                types[] = {"ARRAY"};
+                optional = 0;
+            };
+        };
+    };
+    class civInit {
+        isLocal = 1;
+        class params {
+            class _0 {
+                description = "Unit being initilized";
```
Comment (Bob-Murphy) [inline A3A/addons/Events/Events.hpp:76]: initialized

File: A3A/addons/Events/Events.hpp [hunk 3]
```diff
@@ -1,20 +1,159 @@
...
+                types[] = {"OBJECT"};
+                optional = 0;
+            };
+        };
+    };
+    class civVehInit {
+        isLocal = 1;
+        class params {
+            class _0 {
+                description = "civ vehicle being initilized";
```
Comment (Bob-Murphy) [inline A3A/addons/Events/Events.hpp:86]: initialized

File: A3A/addons/Events/Events.hpp [hunk 4]
```diff
@@ -1,20 +1,159 @@
...
+                types[] = {"BOOL"};
+                optional = 0;
+            };
+        };
+    };
+    class AIInit {
+        isLocal = 1;
+        class params {
+            class _0 {
+                description = "Unit being initilized";
```
Comment (Bob-Murphy) [inline A3A/addons/Events/Events.hpp:116]: initialized

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]
DNLGTM

File: A3A/addons/Events/Events.hpp (hunk 4 above)
Comment (HakonRydland) [reply A3A/addons/Events/Events.hpp]: me no english

**Review by Bob-Murphy — APPROVED** [review]
GNTM

**Pull request merged**